frontend/src/components/menus/TwoFactorComponent.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+    <form @submit="handleSubmit">
+        <div class="user_two_factor_qr">
+            <img v-if="srcImage" id="qrImage" :src="srcImage" class="user_qr_image">
+            <div class="message_two_factor" style="top: 56%">{{ getTittle() }}</div>
+            <div class="message_two_factor">{{ getMessage() }}</div>
+            <input id="inputQR" class="input_Two_Factor" type="text" pattern="[0-9]+" :maxlength="6" required>
+            <div style="top: 90%; left: 10%;">
+                <button type="button" @click="handleCancelClick">Cancel</button>
+                <button style="right: 20%;position: absolute;background-color: white;color: black;"
+                    type="submit">Confirm</button>
+            </div>
+        </div>
+    </form>
+</template>
+
+<script setup lang="ts">
+import { userStore } from '@/stores/userStore';
+import { getCurrentInstance, onMounted, ref } from 'vue';
+
+const instance = getCurrentInstance();
+const srcImage = ref();
+
+function handleSubmit(event: any) {
+    event.preventDefault();
+
+    handleConfirmClick();
+}
+
+async function getQrImage() {
+    try {
+        const data = await userStore().twoFAGenerate();
+        console.log(data);
+        srcImage.value = data;
+    } catch (error) {
+        console.error("Error converting file to base64:", error);
+    }
+}
+
+async function handleConfirmClick() {
+    const inputTwoFactor = document.getElementById("inputQR") as HTMLInputElement;
+
+    try {
+        if (!userStore().user.isTwoFAEnabled) await userStore().twoFATurnOn(inputTwoFactor.value.toString());
+        else await userStore().twoFATurnOff(inputTwoFactor.value.toString());
+
+        inputTwoFactor.value = "";
+        instance?.emit("two-factor-status");
+    } catch (error) {
+        inputTwoFactor.style.borderColor = "red";
+    }
+}
+
+function handleCancelClick() {
+
+    const inputTwoFactor = document.getElementById("inputQR") as HTMLInputElement;
+
+    inputTwoFactor.value = "";
+    instance?.emit("two-factor-status");
+    console.log("Cancel button clicked");
+}
+
+function getTittle() {
+    if (!userStore().user.isTwoFAEnabled)
+        return "To Enable 2FA:"
+    return "To Disable 2FA:";
+}
+
+function getMessage() {
+    if (!userStore().user.isTwoFAEnabled)
+        return "1 - Download the Google Authenticator app from your phone's app store.\n\
+                2 - Launch the app and tap on the '+' icon.\n\
+                3 - Choose \"Scan a QR code\".\n\
+                4 - Scan the QR code displayed above.\n\
+                5 - Enter the generated code into the input box provided and hit \"Confirm\".\n\
+                6 - 2FA is now activated.\n\
+                7 - The next time you log in, you will be prompted to enter your 2FA code.";
+    return "1 - Input your code into the box below and click \"Confirm\".\n\
+            2 - 2FA is now deactivated.";
+}
+
+onMounted(() => {
+    getQrImage();
+});
```

Comment:
Isto não esta correto. Esta a gerar um novo QRCode quando se quer ativar (como é suposto) e esta a gerar outro codigo novo quando se quer desativar (não é suposto). O codigo para desativar tem de ser baseado no QRCode usado na ativação. Isto antes estava bem, se não me engano, mas como tens isso no onMounted, sempre que abre a janela faz a chamada de API para o backend para pedir um novo codigo. A consequencia é que se carregares no botão para desativar e depois cancelares, quando tentares fazer log in de novo, o QRCode lido na ativação não vai ser válido